ArchiSteamFarm/Plugins/PluginsCore.cs

```diff
@@ -766,9 +766,31 @@ public static class PluginsCore {
 
 		try {
 			foreach (string assemblyPath in Directory.EnumerateFiles(path, "*.dll", SearchOption.AllDirectories)) {
-				string? assemblyDirectoryName = Path.GetFileName(Path.GetDirectoryName(assemblyPath));
+				string? assemblyDirectory = Path.GetDirectoryName(assemblyPath);
 
-				if (assemblyDirectoryName == SharedInfo.UpdateDirectory) {
+				if (string.IsNullOrEmpty(assemblyDirectory)) {
+					throw new InvalidOperationException(nameof(assemblyDirectory));
+				}
+
+				// Skip from loading those files that come from update directories
+				// We determine that by checking if any directory name along the path to the assembly matches
+				bool skip = false;
+
+				string? relativeAssemblyDirectory = Path.GetRelativePath(path, assemblyDirectory);
+				string? relativeAssemblyDirectoryName = Path.GetFileName(relativeAssemblyDirectory);
+
+				while (!string.IsNullOrEmpty(relativeAssemblyDirectoryName)) {
+					if ((relativeAssemblyDirectoryName == SharedInfo.UpdateDirectoryOld) || relativeAssemblyDirectoryName.StartsWith(SharedInfo.UpdateDirectoryOldPrefix, StringComparison.Ordinal) || relativeAssemblyDirectoryName.StartsWith(SharedInfo.UpdateDirectoryNewPrefix, StringComparison.Ordinal)) {
```

Comment:
This might break current user setups which use `old-`/`new-` in the plugin directory

Reply:
Yes, but not just `old-` and `new-` but `_old-` and `_new-`. I classify this breaking change as minimal, not to mention this considers only loading dll files - plugin creators can adapt if needed.